VSRAD.Package/Server/CommunicationChannel.cs

```diff
@@ -85,9 +85,21 @@ public CommunicationChannel(SVsServiceProvider provider, IProject project)
                 options.PropertyChanged += (s, e) => { if (e.PropertyName == nameof(options.ActiveProfile)) ForceDisconnect(); };
         }
 
-        public async Task SendAsync(ICommand command)
+        public async Task SendAsync(ICommand command) => await SendAsync(command, true);
+
+        private async Task SendAsync(ICommand command, bool tryReconnect)
         {
-            await EstablishServerConnectionAsync().ConfigureAwait(false);
+            try
```

Comment:
Handling `ConnectionRefusedException` here is not needed, since reconnecting won't help if the destination host is unreachable.

Reply:
This handler is needed to show user "Could not reconnect to server" message in server output window and showing "Connection to {ConnectionOptions} has been terminated:" error instead of "Unable to establish connection..."

Without it it's not that clear what happened from user's point of view. Note that because of `when (!tryReconnect)` check if server is down on a stage of establishing connection, the behavior would be same as before.

Maybe we should consider removing `else` on line 119 instead, since if server is actually down, we'll show error by handling `ConnectionRefusedException` and we'll actually never fall in this case?

Reply:
Okay, let's keep it.

Regarding line 119: the `else` branch is necessary because exceptions there can be unrelated to connection state. The call to `ForceDisconnect()` should be moved out of `else` and placed before the conditional, though. Consider serialization errors: we cannot use the same connection because we may have read the server response only partially, so calling `SendAsync` again without dropping the connection can lead to unexpected errors.

Reply:
Thanks for the suggestion, fixed in 40d8520